contracts/message/libraries/MessageSenderLib.sol

```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
-                    _receiver,
-                    _token,
-                    _amount,
-                    _dstChainId,
-                    _nonce,
-                    _message,
-                    _messageBus,
-                    _fee
-                );
-        } else if (
-            _bridgeSendType == MsgDataTypes.BridgeSendType.PegBurn ||
-            _bridgeSendType == MsgDataTypes.BridgeSendType.PegV2Burn ||
-            _bridgeSendType == MsgDataTypes.BridgeSendType.PegV2BurnFrom
-        ) {
-            return
-                sendMessageWithPegBridgeBurn(
-                    _bridgeSendType,
-                    _receiver,
-                    _token,
-                    _amount,
-                    _dstChainId,
-                    _nonce,
-                    _message,
-                    _messageBus,
-                    _fee
-                );
-        } else {
-            revert("bridge type not supported");
-        }
+        (bytes32 transferId, address bridge) = sendTokenTransfer(
+            _receiver,
+            _token,
+            _amount,
+            _dstChainId,
+            _nonce,
+            _maxSlippage,
+            _bridgeSendType,
+            _messageBus
+        );
+        IMessageBus(_messageBus).sendMessageWithTransfer{value: _fee}(
+            _receiver,
+            _dstChainId,
+            bridge,
+            transferId,
+            _message
+        );
+        return transferId;
     }
 
     /**
-     * @notice Sends a message to an app on another chain via MessageBus with an associated liquidity bridge transfer.
+     * @notice Sends a token transfer via a bridge.
      * @param _receiver The address of the destination app contract.
      * @param _token The address of the token to be sent.
      * @param _amount The amount of tokens to be sent.
      * @param _dstChainId The destination chain ID.
      * @param _nonce A number input to guarantee uniqueness of transferId. Can be timestamp in practice.
      * @param _maxSlippage The max slippage accepted, given as percentage in point (pip). Eg. 5000 means 0.5%.
-     *        Must be greater than minimalMaxSlippage. Receiver is guaranteed to receive at least
-     *        (100% - max slippage percentage) * amount or the transfer can be refunded.
-     * @param _message Arbitrary message bytes to be decoded by the destination app contract.
-     *        If message is empty, only the token transfer will be sent
-     * @param _messageBus The address of the MessageBus on this chain.
-     * @param _fee The fee amount to pay to MessageBus.
-     * @return The transfer ID.
+     * Must be greater than minimalMaxSlippage. Receiver is guaranteed to receive at least (100% - max slippage percentage) * amount or the
+     * transfer can be refunded.
+     * @param _bridgeSendType One of the {MsgDataTypes.BridgeSendType} enum.
      */
-    function sendMessageWithLiquidityBridgeTransfer(
+    function sendTokenTransfer(
```

Comment:
looks similar to https://github.com/celer-network/sgn-v2-contracts/blob/main/contracts/libraries/BridgeTransferLib.sol#L64

Reply:
I only extracted the "token transfer" part from sendMessageWithXXX functions. We could use BridgeTransferLib in MessageSenderLib, but doing this requires the user of this lib to also copy 3 XXX.Pb.sol files, even if they don't use those